crates/router/src/connector/airwallex.rs

```diff
@@ -962,9 +962,37 @@ impl api::IncomingWebhook for Airwallex {
             .parse_struct("airwallexWebhookData")
             .change_context(errors::ConnectorError::WebhookReferenceIdNotFound)?;
 
-        Ok(api_models::webhooks::ObjectReferenceId::PaymentId(
-            api_models::payments::PaymentIdType::ConnectorTransactionId(details.source_id),
-        ))
+        if airwallex::is_transaction_event(&details.name) {
+            return Ok(api_models::webhooks::ObjectReferenceId::PaymentId(
+                api_models::payments::PaymentIdType::ConnectorTransactionId(
+                    details
+                        .source_id
+                        .ok_or(errors::ConnectorError::WebhookReferenceIdNotFound)?,
+                ),
+            ));
+        }
+        if airwallex::is_refund_event(&details.name) {
+            return Ok(api_models::webhooks::ObjectReferenceId::RefundId(
+                api_models::webhooks::RefundIdType::ConnectorRefundId(
+                    details
+                        .source_id
+                        .ok_or(errors::ConnectorError::WebhookReferenceIdNotFound)?,
+                ),
+            ));
+        }
+        if airwallex::is_dispute_event(&details.name) {
+            let dispute_details: airwallex::AirwallexDisputeObject = details
+                .data
+                .object
+                .parse_value("AirwallexDisputeObject")
+                .change_context(errors::ConnectorError::WebhookBodyDecodingFailed)?;
+            return Ok(api_models::webhooks::ObjectReferenceId::PaymentId(
+                api_models::payments::PaymentIdType::ConnectorTransactionId(
+                    dispute_details.payment_intent_id,
+                ),
+            ));
+        }
+        Err(errors::ConnectorError::WebhookReferenceIdNotFound).into_report()
```

Comment:
We can avoid the return statement here. Also the error type is not appropriate, If we are not able to match the event type to any of the required types, then we should not raise `WebhookReferenceIdNotFound`. It should be `UnsupportedWebhookEventType`.